src/main/java/org/broadinstitute/hellbender/tools/exome/copyratio/CopyRatioData.java

```diff
@@ -0,0 +1,112 @@
+package org.broadinstitute.hellbender.tools.exome.copyratio;
+
+import com.google.common.primitives.Doubles;
+import org.apache.commons.math3.stat.descriptive.moment.Mean;
+import org.apache.commons.math3.stat.descriptive.moment.Variance;
+import org.broadinstitute.hellbender.tools.exome.ReadCountRecord;
+import org.broadinstitute.hellbender.tools.exome.SegmentedGenome;
+import org.broadinstitute.hellbender.tools.exome.TargetCollection;
+import org.broadinstitute.hellbender.utils.mcmc.DataCollection;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * {@link DataCollection} for the copy-ratio model containing the set of coverages and the grouping of coverages into segments.
+ *
+ * @author Samuel Lee &lt;slee@broadinstitute.org&gt;
+ */
+public final class CopyRatioData implements DataCollection {
+    private final int numSegments;
+    private final int numTargets;
+    private final double coverageMin;
+    private final double coverageMax;
+    private final double coverageRange;
+
+    private final List<List<Double>> coveragesPerSegment = new ArrayList<>();
+    private final List<Integer> numTargetsPerSegment;
+    private final List<Integer> startTargetsPerSegment = new ArrayList<>();
+
+    public CopyRatioData(final SegmentedGenome segmentedGenome) {
+        if (segmentedGenome.getGenome().getTargets().targetCount() == 0) {
+            throw new IllegalArgumentException("Cannot construct CopyRatioData with no target-coverage data.");
+        }
+        //construct coverages and number of targets per segment from the segments and Genome in SegmentedGenome
+        final TargetCollection<ReadCountRecord.SingleSampleRecord> targetCoverages = segmentedGenome.getGenome().getTargets();
+        //construct list of coverages (in order corresponding to that of segments in SegmentedGenome;
+        //this may not be in genomic order, depending on how the segments are sorted in the segment file,
+        //so we cannot simply take the list of coverages in the order from TargetCollection.targets()
+        final List<Double> coverages = segmentedGenome.getSegments().stream()
+                .flatMap(s -> targetCoverages.targets(s).stream())
+                .map(ReadCountRecord.SingleSampleRecord::getCount)
+                .collect(Collectors.toList());
+        numTargetsPerSegment = segmentedGenome.getSegments().stream()
+                .map(s -> targetCoverages.targets(s).size())
+                .collect(Collectors.toList());
+        numSegments = numTargetsPerSegment.size();
+        numTargets = coverages.size();
+        coverageMin = Collections.min(coverages);
+        coverageMax = Collections.max(coverages);
+        coverageRange = coverageMax - coverageMin;
+        //partition coverages by segment
```

Comment:
i instead refactored by storing coverages + target indices partitioned by segment instead.  cleans things up in a few places, i think.